### PR TITLE
Add slice method to ENV like Hash#slice

### PR DESCRIPTION
**Pull request by benoittgt**
Hello

After reading https://bugs.ruby-lang.org/issues/14559 I thought it could be a good idea to try to implement this during the RubyHackChallenge at Cookpad office in Bristol. 

Here is my first PR for MRI, feel free to make any comments.
I don't know if I need to add entry changelog somewhere.

Thanks @mame for the help when I was working on.

### CODE REVIEW

**Comment by nobu** [comment]
Could you adjust the indent?

**Comment by benoittgt** [comment]
Sorry. Is it better now @nobu ?

Thanks for the review

**Comment by benoittgt** [comment]
Hello @nobu 
Sorry to ping you again.

File: test/ruby/test_env.rb [hunk 1]
```diff
@@ -281,6 +281,17 @@ def test_filter
     end
   end
 
+  def test_slice
+    ENV.clear
+    ENV["foo"] = "bar"
+    ENV["baz"] = "qux"
+    ENV["bar"] = "rab"
+    assert_equal(ENV.slice(), {})
```
Comment (nobu) [inline test/ruby/test_env.rb]: The order of `assert_equal` is (expected, result).

**Comment by nobu** [comment]
Seems fine, except for the arguments order of assertions.
I think that this would be acceptable.

**Comment by benoittgt** [comment]
Sorry for the arguments order of assertions. It should be good now.

I have one failing test on AppVeyor but it seems unrelated

```
# Running tests:
running file: C:/projects/ruby/test/open-uri/test_open-uri.rb
Some worker was crashed. It seems ruby interpreter's bug
or, a bug of test/unit/parallel.rb. try again without -j
option.
NMAKE : fatal error U1077: '.\ruby.exe' : return code '0x1'
Stop.
Command exited with code 2
```

**Comment by benoittgt** [comment]
Closed with https://github.com/ruby/ruby/commit/9760a7f905367e96d6c22bbace1680d694fa16bd

Thanks @nobu

**Pull request closed without merging**